### PR TITLE
shibboleth-sp: fix build on darwin

### PR DESCRIPTION
**Pull request by eliandoran**
## Things done

Added missing darwin frameworks and added workaround for `auto_ptr`.

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by eliandoran** [comment]
Result of `nixpkgs-review pr 312686` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

File: pkgs/development/libraries/shibboleth-sp/default.nix [hunk 1]
```diff
@@ -20,7 +24,7 @@ stdenv.mkDerivation rec {
     "--with-fastcgi"
   ];
 
-  env.NIX_CFLAGS_COMPILE = lib.optionalString (!stdenv.isDarwin) "-std=c++14";
+  env.NIX_CFLAGS_COMPILE = if stdenv.isDarwin then "-D_LIBCPP_ENABLE_CXX17_REMOVED_AUTO_PTR" else "-std=c++14";
```
Comment (wegank) [inline pkgs/development/libraries/shibboleth-sp/default.nix:27]: Looks like `"-std=c++14"` might work too? Or are there other problems?

Comment (eliandoran) [reply pkgs/development/libraries/shibboleth-sp/default.nix:27]: @wegank , I've thought so as well, but it seems not to work on darwin. More specifically, `configure` fails early with:

```
checking whether the C compiler works... no
configure: error: in '/private/tmp/nix-build-shibboleth-sp-3.0.4.1.drv-0/cpp-sp':
configure: error: C compiler cannot create executables
See 'config.log' for more details
```

Comment (annaleeleaves) [reply pkgs/development/libraries/shibboleth-sp/default.nix:27]: `-std=c++14` isn't a valid flag for a c compiler, thus configure fails. removing `env.NIX_CFLAGS_COMPILE` and instead adding `"CXXFLAGS=-std=c++14"` to the `configureFlags` should be sufficient.

Comment (eliandoran) [reply pkgs/development/libraries/shibboleth-sp/default.nix]: @annaleeleaves , thank you, works like a charm.

**Review by ghost — APPROVED** [review]
LGTM. builds fine on x64 linux / darwin

**Pull request merged**